src/main/java/com/revealprecision/revealserver/service/MetaDataJdbcService.java

```diff
@@ -0,0 +1,142 @@
+
+@Service
+public class MetaDataJdbcService {
+
+  @Autowired
+  JdbcTemplate jdbcTemplate;
+
+  @Autowired
+  BusinessStatusProperties businessStatusProperties;
+
+  @Autowired
+  EntityTagService entityTagService;
+
+
+  public Map<String, Pair<Class<?>, Object>> getMetadataFor(String entityTable, UUID identifier) {
+
+    String metadataTable = entityTable + "_metadata";
+
+    String query = "SELECT t.key,t.value->>'value',t.value->>'entity_tag',et.value_type From (\n"
+        + "SELECT  (jsonb_each(lm.entity_value)).*  from " + entityTable + " l\n" + "inner join "
+        + metadataTable+" lm on l.identifier = lm." + entityTable + "_identifier\n"
+        + "where l.identifier = '" + identifier + "'\n"
+        + "\t) as t left join entity_tag et on et.identifier = (t.value->>'entity_tag')::uuid";
+
+    return jdbcTemplate.query(query, (rs, rowNum) -> getMapFromResultSet(rs)).stream()
+        .filter(Objects::nonNull).flatMap(mapOfKeyAndPair -> mapOfKeyAndPair.entrySet().stream())
+        .collect(Collectors.toMap(Entry::getKey, Entry::getValue));
+
+  }
+
+  public void createOrUpdateMetadata(UUID entityIdentifier, String entity, String tagName,
+      Object value, Class<?> entityDataType) {
+    String quotes = (entityDataType.equals(String.class) ? "'" : "");
+
+
+// upsert String resolves to below insert statement
+
+//    INSERT INTO location_metadata(
+//        identifier, entity_value, location_identifier, entity_status, created_by, created_datetime, modified_by, modified_datetime)
+//    values
+//        (uuid_generate_v4()
+//            ,(jsonb_build_object(
+//                'business_status'
+//                ,(jsonb_build_object(
+//                    'entity_tag'
+//                    ,(SELECT identifier from entity_tag where tag = 'business_status'
+//                    AND lookup_entity_type_identifier = (SELECT identifier FROM lookup_entity_type WHERE table_name = 'location'))
+//                ,'value'
+//                ,'NOT_VISITED'
+//            ))
+//        ))
+//	  ,'76a83898-efec-4470-921e-4834d8245c7d'
+//        , 'ACTIVE'
+//        , '649f338b-eb53-4832-9562-f695e9cc44e7'
+//        , '2022-01-12 13:54:22.106221+00'
+//        , '649f338b-eb53-4832-9562-f695e9cc44e7'
+//        , '2022-01-12 13:54:22.106221+00')
+//    ON CONFLICT (location_identifier) DO UPDATE
+//    SET entity_value =
+//        jsonb_set(
+//            location_metadata.entity_value
+//            ,'{business_status}'
+//            ,(jsonb_build_object(
+//                'entity_tag'
+//                ,(SELECT identifier from entity_tag where tag = 'business_status' AND lookup_entity_type_identifier =
+//                    (SELECT identifier FROM lookup_entity_type WHERE table_name = 'location'))
+//            ,  'value','NOT_VISITED'
+//        ))
+//				,true
+//			),
+//    WHERE location_metadata.location_identifier = '76a83898-efec-4470-921e-4834d8245c7d'
+
+    String metadataTable = entity + "_metadata";
```

Comment:
I'm seeing this for the second time, can be extracted to constant

Reply:
will consider